src/Engine/Component/ParticleComponents/ParticleComponent.cpp

```diff
@@ -0,0 +1,306 @@
+#include "ParticleComponent.hpp"
+
+ParticleComponent::ParticleComponent(GameObject & gameobject) :
+    Component(gameobject)
+{  
+}
+
+ParticleComponent::ParticleComponent(GameObject & gameobject, SpatialComponent * anchor) :
+    Component(gameobject),
+    m_anchor(anchor)
+{
+}
+
+ParticleComponent::ParticleComponent(GameObject & gameobject, SpatialComponent * anchor, SpatialComponent * directional) :
+    Component(gameobject),
+    m_anchor(anchor),
+    m_directional(directional)
+{
+}
+
+void ParticleComponent::init() {
+    const glm::fmat3 & rotation = glm::fmat3();
+    const glm::fvec3 & translation = glm::fvec3(0.0f);
+    const glm::fvec3 & scale = glm::fvec3(getScaleFactor(m_effect));
+    glm::mat4 comp = Util::compositeTransform(scale, translation);
+    m_M = comp;
+    m_N = glm::mat3();
+
+    initRandomOrientations(comp, m_N, m_activeEffect->getOrientationCount());
+}   
+
+void ParticleComponent::update(float dt) {
+    if (m_anchor != NULL) {
+        if (m_directional == NULL) {
+            m_activeEffect->update(dt, m_anchor->position());
+        }
+        else {
+            m_activeEffect->update(dt, m_anchor->position(), -m_directional->w());
+        }
+
+    }
+    else {
+        m_activeEffect->update(dt);
+    }
+}
+
+void ParticleComponent::spawnParticleEffect(ParticleEffect::Effect effect, const glm::vec3 & position, const glm::vec3 & direction, 
+    const glm::vec3 & velocity) {
+    ParticleEffect::EffectParams *effectParams = getEffectParams(effect);
+    m_activeEffect = new ParticleEffect(effectParams, position, direction, velocity);
```

Comment:
I trust that these new calls are being deleted when the particle dies?

Reply:
At the end of the particle Effect. I'll add this to Particle System.